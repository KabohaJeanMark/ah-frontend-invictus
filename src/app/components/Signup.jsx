import React, { Component } from "react";
import "../../assets/styles/signup.scss";
import { Link } from "react-router-dom";

class SignupFormComponent extends Component {
  render () {
    const {
      signupResponse,
      signupErrors,
      handleChange,
      handleSignup,
      timeMessage,
      isLoading
    } = this.props;
    return (
      <React.Fragment>
        <div className="row">
          <div className="col-md-4" />
          <div className="col-md-4">
            <div className="banner">
              <img className="banner-img"
                src={require("../../assets/images/ah_banner.png")}
                alt="Banner" />
            </div>
            {Object.keys(signupResponse).length > 0 ? (<div id="success" className="success">
              {timeMessage()}
              <p>Account was created. checkyour email to activate it.</p></div>) : ("")}
            <div className="card">
              <form id="signup-form"
                onSubmit={handleSignup}>
                <input
                  onChange={handleChange}
                  type="text"
                  id="username"
                  placeholder="username"
                  className="form-control input-field"

                />
                <p>{signupErrors.username ? (<span className="input-error" id="username_error">
                  {signupErrors.username}</span>) : ("")}</p>
                <input
                  onChange={handleChange}
                  type="text"
                  id="email"
                  placeholder="email"
                  className="form-control input-field"
                />
                <p>{signupErrors.email ? (<span className="input-error" id="email_error">
                  {signupErrors.email}</span>) : ("")}</p>
                <input
                  onChange={handleChange}
                  type="password"
                  id="password"
                  placeholder="password"
                  className="form-control input-field"
                />
                <p>{signupErrors.password ? (<span className="input-error"
                  id="password_error">{signupErrors.password}</span>) : ("")}</p>
                {isLoading ? (<button className="signup-btn"
                  type="submit" disabled>
                  <span className="spinner-border spinner-border-sm" />
                  Signing up..</button>) : (<button className="signup-btn"
                  type="submit">Signup</button>)}
                <div className="signupwith">
                  <img className="signupwith-img" alt="signup-with"
                    src={require("../../assets/images/signupwith.png")} />
                  <div className="row social-media-row">
                    <div className="col-md-4 col-2">
                      <img className="socialmedia-img-signup" alt="facebook-logo"
                        src={require("../../assets/images/facebook.png")} />
                    </div>
                    <div className="col-md-4 col-2">
                      <img className="socialmedia-img-signup twitter-img" alt="twitter-logo"
                        src={require("../../assets/images/twitter.png")} />
                    </div>
                    <div className="col-md-4 col-2">
                      <img className="socialmedia-img-signup" alt="google-logo"
                        src={require("../../assets/images/google.png")} />
                    </div>
                  </div>
                </div>
                <p className="have-an-account">Already have an account!
                  <Link to="/login">Login</Link></p>
              </form>
            </div></div>
          <div className="col-md-4" />
        </div>
      </React.Fragment>
    );
  }
}

export default SignupFormComponent;
