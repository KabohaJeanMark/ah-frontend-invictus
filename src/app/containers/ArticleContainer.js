import React, { Component } from 'react';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import ArticlePresentationComponent from '../components/ArticleComponent';
import { postArticle } from '../../redux/actions/ArticleActions';
import { articleUrl } from '../utils';

export class ArticleContainer extends Component {
  state = {
    title: "",
    description: "",
    body: "",
    tagList: ["tags"]
  };

  checkBlank = () => {
    const { title, body, description } = this.state;
    if (title === '' || body === '' || description === '') {
      toast.dismiss();
      toast.error("All fields must be filled", {
        hideProgressBar: false,
        autoClose: 3000
      });
    }
  };

  handleChange = (event) => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };
  handleSubmit = (event) => {
    event.preventDefault();
    const { title, body, description, tagList } = this.state;
    const articleDetails = { title, body, description, tagList };
    this.checkBlank();
    var headers = {};
    headers['Content-Type'] = 'application/json';
    headers['Authorization'] = `Bearer ${localStorage.getItem('user_token')}`;
    this.props.postArticle(articleUrl, articleDetails, headers);
  };

  render () {
    return (

      <ArticlePresentationComponent
        handleSubmit={this.handleSubmit}
        handleChange={this.handleChange}
      />
    );
  }
}

export default connect(null, { postArticle })(ArticleContainer);
