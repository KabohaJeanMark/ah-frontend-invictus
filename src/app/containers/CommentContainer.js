import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import Comment from '../../app/components/Comment';
import { fetchComments } from '../../redux/actions/FetchCommentAction';
import { postComment } from '../../redux/actions/CommentAction';
import FetchComment from '../../app/components/FetchComment';

export class CommentContainer extends Component {
  constructor (props) {
    super(props);

    this.state = {
      commentBody: "",
      displayComment: []
    };
  }
  componentDidMount () {
    const { fetchComments, articleId } = this.props;
    fetchComments(articleId);
  }

  componentWillReceiveProps = (nextProps) => {
    if (nextProps.comments) {
      this.setState({ displayComment: nextProps.comments.comments });
    }
  }

  emptyBody = (comment) => {
    if (comment === undefined) {
      toast.dismiss();
      toast.error("The comment body is required", {
        hideProgressBar: false,
        autoClose: 5000
      });
    }
  };

  handleChange = (event) => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  handleSubmit = (event) => {
    const { articleId } = this.props;
    event.preventDefault();
    const { commentBody } = this.state;
    var headers = {};
    headers['Content-Type'] = 'application/json';
    headers['Authorization'] = `Bearer ${localStorage.getItem('user_token')}`;
    this.emptyBody(commentBody);
    this.props.postComment(articleId, commentBody);
  };

  render () {
    const { comments } = this.props;
    return (

      <React.Fragment>
        <Comment
          handleSubmit={this.handleSubmit}
          handleChange={this.handleChange}
        />
        <FetchComment comments={comments} />
      </React.Fragment>
    );
  }
}

export const mapStateToProps = state => ({
  comment: state.postComment,
  comments: state.fetchComment.comments
});

CommentContainer.propTypes = {
  fetchComment: PropTypes.func
};

export default connect(mapStateToProps, { postComment, fetchComments })(CommentContainer);
