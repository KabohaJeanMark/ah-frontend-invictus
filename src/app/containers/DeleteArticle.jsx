import React, { Component } from "react";
import { connect } from "react-redux";
import { deleteArticleAction } from '../../redux/actions/DeleteArticle';
import DeleteArticleComponent from '../components/DeleteArticle';

export class DeleteArticleContainer extends Component {
  handleDelete = event => {
    /* eslint-disable no-alert*/
    const delArticle = window.confirm("Are you sure you want to delete this article");
    if (delArticle === true) {
      this.props.deleteArticleAction(this.props.id);
    }
  }
  render () {
    return (
      <>
        < DeleteArticleComponent
          handleDelete={this.handleDelete}
        />
      </>
    );
  }
}

export const mapStateToProps = state => ({
  deleteArticleErrors: state.deleteArticle.deleteArticleErrors,
  deleteArticleResponse: state.deleteArticle.deleteArticleResponse
});

export default connect(
  mapStateToProps,
  { deleteArticleAction }
)(DeleteArticleContainer);
