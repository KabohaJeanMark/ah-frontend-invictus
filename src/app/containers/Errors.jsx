import React, { Component } from 'react';

export default class Errors extends Component {
  constructor (props) {
    super(props);
    this.state = { errors: [] };
  }
  componentWillReceiveProps (nextProps) {
    if (nextProps.errors) {
      this.setState({
        errors: this.mapErrorsToList(nextProps)
      });
    }
  }
   mapErrorsToList = props => {
     let items = [];
     if (Array.isArray(props.errors)) {
       for (let i = 0; i < props.errors.length; i++) {
         items.push(props.errors[i]);
       }
     } else {
       Object.keys(props.errors).forEach(key => {
         for (let i = 0; i < props.errors[key].length; i++) {
           items.push(key + ': ' + props.errors[key][i]);
         }
       });
     }
     return items;
   }
   render () {
     function Repeat (props) {
       let items = [];
       for (let i = 0; i < props.errors.length; i++) {
         items.push(props.children(i, props.errors[i]));
       }
       return <div className="danger" ><ul>{items}</ul></div>;
     }
     return (
       this.state.errors.length ? (
         <Repeat errors={this.state.errors}>
           {(index, message) => <li key={index}>{message}</li>}
         </Repeat>
       ) : (null)
     );
   }
}
