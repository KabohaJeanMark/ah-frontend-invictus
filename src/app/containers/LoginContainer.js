import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { loginAction } from '../../redux/actions/LoginActions';
import Login from '../components/Login';
import { loginUtil } from './loginUtils';

export class LoginContainer extends Component {
  constructor () {
    super();
    this.state = {
      email: '',
      password: '',
      errors: {}
    };
  }
  onChange = (event) => {
    event.preventDefault();
    this.setState({ [event.target.name]: event.target.value });
  }
  checkValidEmail = (userEmail) => {
    const mailFormat = loginUtil.LOGIN_UTILS;
    if (userEmail.match(mailFormat)) {
      return true;
    }
    return false;
  };

  checkPassword = (userPassword) => {
    if (userPassword === undefined) {
      return false;
    }
    return true;
  };

  handleSubmit = (event) => {
    event.preventDefault();
    const userData = {
      user: {
        email: this.state.email,
        password: this.state.password
      }
    };
    if (this.state.email === undefined) {
      this.setState({ errors: { userEmail: 'Email required' } });
    } else if (!this.checkValidEmail(this.state.email)) {
      this.setState({ errors: { userEmail: 'invalid email format' } });
    } else if (!this.checkPassword(this.state.password)) {
      this.setState({ errors: { userPassword: 'Password field is empty' } });
    } else {
      this.props.loginAction(userData, this.props);
    }
  }
  render () {
    return (
      <Login
        handleSubmit={this.handleSubmit}
        onChange={this.onChange}
        email={this.state.email}
        password={this.state.password}
        errors={this.state.errors}
      />
    );
  }
}

LoginContainer.propTypes = {
  email: PropTypes.string,
  password: PropTypes.string,
  handleSubmit: PropTypes.func,
  onChange: PropTypes.func,
  loginAction: PropTypes.func,
  errors: PropTypes.object
};

LoginContainer.defaultProps = {
  errors: {},
  email: '',
  password: '',
  handleSubmit: () => { },
  onChange: () => { }
};
export const mapStateToProps = state => ({ login: state.auth });
export default connect(mapStateToProps, { loginAction })(LoginContainer);
