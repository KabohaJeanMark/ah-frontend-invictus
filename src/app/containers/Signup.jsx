import React, { Component } from "react";
import { connect } from "react-redux";
import { signupUserAction } from "../../redux/actions/Signupactions";
import SignupFormComponent from "../components/Signup";
import NavbarComponent from "../components/Navbar";

export class SignupComponent extends Component {
  constructor (props) {
    super(props);
    this.state = {
      username: "",
      email: "",
      password: ""
    };
  }
  handleSignup = event => {
    event.preventDefault();
    let form = document.getElementById("signup-form");
    const { username, email, password } = this.state;
    const signData = {
      username: username,
      email: email,
      password: password
    };
    if (form) {
      form.reset();
    }
    this.props.signupUserAction(signData);
  }
  handleChange = event => {
    this.setState({ [event.target.id]: event.target.value });
  }

  timeMessage = () => {
    setTimeout(function () {
      let element = document.getElementById("success");
      if (element) {
        element.style.display = "none";
      }
    }, 10000);
  }
  render () {
    const { signupErrors, signupResponse, isLoading } = this.props;
    return (
      <div>
        <NavbarComponent />
        <SignupFormComponent
          signupResponse={signupResponse}
          signupErrors={signupErrors}
          handleChange={this.handleChange}
          handleSignup={this.handleSignup}
          timeMessage={this.timeMessage}
          isLoading={isLoading}
        />
      </div >
    );
  }
}

export const mapStateToProps = state => ({
  signupErrors: state.signup.signupErrors,
  signupResponse: state.signup.signupResponse,
  isLoading: state.signup.isLoading
});

export default connect(
  mapStateToProps,
  { signupUserAction }
)(SignupComponent);
