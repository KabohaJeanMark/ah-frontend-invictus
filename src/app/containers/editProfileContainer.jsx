import React, { Component } from 'react';
import { connect } from 'react-redux';
import firebase from '../../firebase/config';
import { editProfile } from '../../redux/actions/profileActions';
import '../../assets/styles/userprofile.scss';
import NavbarComponent from '../components/Navbar';

export class EditProfileContainer extends Component {
  constructor (props) {
    super(props);
    this.state = {
      firstname: "",
      lastname: "",
      bio: "",
      image: "",
      progress: 0,
      uploadDone: false
    };
  }

  componentWillMount () {
    const { firstname, lastname, bio, image } = this.props.profile.profile;
    this.setState({ firstname, lastname, bio, image });
  }

  componentWillReceiveProps (nextProps) {
    if (nextProps.editSuccess === true) {
      const { history } = nextProps;
      history.push('/profile');
    }
  }

  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }
  onSubmit = (e) => {
    e.preventDefault();
    const { firstname, lastname, bio, image } = this.state;
    const newProfile = {
      firstname: firstname,
      lastname: lastname,
      bio: bio,
      image: image
    };
    this.props.editProfile(newProfile, this.props);
  }

  /* istanbul ignore next */
  uploadImage (files) {
    const fileload = firebase
      .storage()
      .ref(`images/${files[0].name}`)
      .put(files[0]);

    fileload.then(res => {
      firebase
        .storage()
        .ref(`images/${files[0].name}`)
        .getDownloadURL()
        .then(url => {
          const image = {
            image: url
          };
          this.props.editProfile(image);
        });
    });
    fileload.on("state_changed", snapshot => {
      const uploadDone = true;
      this.setState({ uploadDone });
      const progress = Math.round(
        (snapshot.bytesTransferred / snapshot.totalBytes) * 100
      );
      this.setState({ progress });
    });
  }

  render () {
    return (
      <div>
        <div className="row">
          <div className="col-md-12"><NavbarComponent /></div>
        </div>
        <div className="row">
          <div className="col-md-4" />
          <div className="col-md-4">
            <div className="image-div" >
              {this.state.image === "" ? (<img alt="profileimage" className="profile-image"
                src={require("../../assets/images/avatar2.png")} />) :
                (<img className="profile-image" alt="profileimage" src={this.state.image} />)}
              <progress value={this.state.progress} max="100" />
              <input
                type="file"
                onChange={event => this.uploadImage(event.target.files)}
                className="upload-button mt-3"
                id="profile"
              />
            </div>
            <form onSubmit={this.onSubmit}>
              <div>
                <div className="form-group">
                  <label className="">firstname</label>
                  <input type="text" name="firstname"
                    onChange={this.onChange} className="form-control"
                    defaultValue={this.state.firstname} />
                </div>
                <div className="form-group">
                  <label className="">Lastname</label>
                  <input type="text" name="lastname" className="form-control"
                    onChange={this.onChange}
                    defaultValue={this.state.lastname} />
                </div>
                <div className="form-group">
                  <label className="">Bio</label>
                  <textarea className="form-control" name="bio" id="exampleFormControlTextarea1"
                    rows="3" onChange={this.onChange} value={this.state.bio} />
                </div>
              </div>
              <div><button className="button" type="submit">Save Changes</button></div>
            </form>
          </div>
          <div className="col-md-4" />
        </div>
      </div>
    );
  }
}


export const mapStateToProps = state => ({
  profile: state.profileReducer,
  editSuccess: state.profileReducer.editSuccess
});

export default connect(mapStateToProps, { editProfile })(EditProfileContainer);
