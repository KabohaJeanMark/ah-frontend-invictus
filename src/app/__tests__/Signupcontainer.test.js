import React from 'react';
import { shallow } from "enzyme";
import { SignupComponent, mapStateToProps } from "../containers/Signup";

describe('<SignupComponent/>', () => {
  let wrapper;
  const props = {};
  beforeEach(() => {
    wrapper = shallow(<SignupComponent {...props} />);
  });
  it("should render SignupComponent without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should test the handleChange event", () => {
    const event = {
      target: {
        id: "username",
        value: "sanya"
      }
    };
    wrapper.instance().handleChange(event);
    expect(wrapper.instance().state.username).toBe("sanya");
  });
  it("should handle signUp event", () => {
    const props = {
      signupUserAction: jest.fn()
    };
    let wrapper = shallow(<SignupComponent {...props} />);
    const instance = wrapper.instance();
    instance.handleSignup({ preventDefault: jest.fn() });
    expect(props.signupUserAction).toHaveBeenCalled();
  });
  it("should map state to props", () => {
    const initialState = {
      signup: {
        signupErrors: {},
        signupResponse: {
        },
        isLoading: false
      }
    };
    expect(mapStateToProps(initialState)).toEqual({
      signupErrors: {},
      signupResponse: {
      },
      isLoading: false
    });
  });
  it("should call the time message method", () => {
    const props = {
      timeMessage: jest.fn(),
      reset: jest.fn(),
      setTimeout: jest.fn()
    };
    let wrapper = shallow(<SignupComponent {...props} />);
    wrapper.instance().timeMessage();
  });
});
