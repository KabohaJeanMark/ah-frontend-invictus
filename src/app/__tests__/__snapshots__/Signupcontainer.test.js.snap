// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupComponent/> should render SignupComponent without crashing 1`] = `ShallowWrapper {}`;
